packages/smart-contracts/src/contracts/ERC20TransferableReceivable.sol

```diff
@@ -183,60 +184,10 @@ contract ERC20TransferableReceivable is ERC721, ERC721Enumerable, ERC721URIStora
     receivableInfoMapping[currentReceivableTokenId] = ReceivableInfo({
       tokenAddress: erc20Addr,
       amount: amount,
-      balance: 0
+      balance: 0,
+      requestID: requestID
     });
 
     _mint(owner, currentReceivableTokenId);
-    _setTokenURI(currentReceivableTokenId, newTokenURI);
-  }
-
-  /**
-   * @notice Get an array of all receivable token IDs owned by a specific address.
-   * @param _owner The address that owns the receivable tokens.
-   * @return An array of all receivable token IDs owned by the specified address.
-   */
-  function getTokenIds(address _owner) public view returns (uint256[] memory) {
-    uint256[] memory _tokensOfOwner = new uint256[](ERC721.balanceOf(_owner));
-    uint256 i;
-
-    for (i = 0; i < ERC721.balanceOf(_owner); i++) {
-      _tokensOfOwner[i] = ERC721Enumerable.tokenOfOwnerByIndex(_owner, i);
-    }
-    return (_tokensOfOwner);
-  }
-
-  // The following functions are overrides required by Solidity.
-  /// @dev Overrides ERC721's _beforeTokenTransfer method to include functionality from ERC721Enumerable.
-  function _beforeTokenTransfer(
-    address from,
-    address to,
-    uint256 tokenId
-  ) internal override(ERC721, ERC721Enumerable) {
-    super._beforeTokenTransfer(from, to, tokenId);
-  }
-
-  /// @dev Overrides ERC721's _burn method to include functionality from ERC721URIStorage.
-  function _burn(uint256 tokenId) internal override(ERC721, ERC721URIStorage) {
-    super._burn(tokenId);
-  }
-
-  /// @dev Overrides ERC721's tokenURI method to include functionality from ERC721URIStorage.
-  function tokenURI(uint256 tokenId)
-    public
-    view
-    override(ERC721, ERC721URIStorage)
-    returns (string memory)
-  {
-    return super.tokenURI(tokenId);
-  }
-
-  /// @dev Overrides ERC721's supportsInterface method to include functionality from ERC721Enumerable.
-  function supportsInterface(bytes4 interfaceId)
-    public
-    view
-    override(ERC721, ERC721Enumerable)
-    returns (bool)
-  {
-    return super.supportsInterface(interfaceId);
   }
 }
```

Comment:
Change Request: 

On [line 118](https://github.com/RequestNetwork/requestNetwork/pull/1097/files#diff-58d3632849e76bbd7f443b5978ff1800d7e3bbf7bd87b13d8152b238107229c2R118), the `paymentReference` argument is `bytes` but it could be restricted to `bytes8` which should save some gas. [ERC20FeeProxy Spec for reference](https://github.com/RequestNetwork/requestNetwork/blob/6ac8c287b8838af93e9384bc30352b334dba116a/packages/advanced-logic/specs/payment-network-erc20-fee-proxy-contract-0.1.0.md?plain=1#L14)
```solidity
bytes calldata paymentReference,
```

Reply:
I now know that other payment proxy smart contracts use `bytes` for the `paymentReference` but I don't understand why.

Reply:
No idea! I saw that as well and kept it as `bytes8` to stay consistent but paymentReference could definitely fit in a `bytes8`. What do you think I should use?

Reply:
Would it be possible to benchmark how much gas is saved by using `bytes8` vs `bytes`? Consistency with other payment proxies has value but I want to understand how much we gain if we break that consistency.

Reply:
Why did the payOwner # of calls increase from 10 to 11 between the tests?

Reply:
One of the tests was to check that payOwner reverted if an empty bytes array was passed in, but I removed this test for the bytes8 benchmark. I don't think this would impact the benchmarks enough to warrant changing over to bytes8 though

Reply:
Thanks for explaining. I agree that switching the `paymentReference` to `bytes8` doesn't make sense right now. It might make sense later if we migrated to a new version of the `ERC20FeeProxy` that also uses `bytes8`.

Reply:
I learned from @vrolland that there's no good reason why `bytes` was used instead of `bytes8`. It just wasn't considered at the time. But now that all the payment proxies follow the same pattern, changing it would mean deploying new versions and migrating. I created #1113 to track this issue.